src/PythonMigrationViewExtension/PythonMigrationAssistantViewModel.cs

```diff
@@ -23,7 +27,25 @@ private void MigrateCode()
 
         public void ChangeCode()
         {
+            SavePythonMigrationBackup();
             PythonNode.MigrateCode(NewCode);
         }
+
+        private void SavePythonMigrationBackup()
+        {
+            var path = GetPythonMigrationBackupPath();
+            if (File.Exists(path))
+                return;
+
+            DynamoViewModel.SaveAs(path, true);
+        }
+
+        private string GetPythonMigrationBackupPath()
+        {
+            var workspaceName = DynamoViewModel.CurrentSpace.Name;
+            var backupDirectory = DynamoViewModel.Model.PathManager.BackupDirectory;
+            var fileName = Path.Combine(backupDirectory, workspaceName) + string.Concat(".", Properties.Resources.PythonMigrationBackupExtension, ".dyn");
```

Comment:
I guess for custom nodes we would want to keep the extension `dyf`.

Reply:
The pathmanager can already be accessed without using the viewModel !!

Reply:
added handling for `dyf` in e9c4d5506205e7b494eecbde2b12d13eab902a7a, will look at `PathManager` next